src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildPropertyParser.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+#nullable enable
+
+namespace Microsoft.DotNet.Cli.Utils;
+
+public static class MSBuildPropertyParser {
+    public static IEnumerable<(string key, string value)> ParseProperties(string input) {
+        var currentPos = 0;
+        string? currentKey = null;
+        string? currentValue = null;
+        
+        (string key, string value) EmitAndReset() {
+            var key = currentKey!;
+            var value= currentValue!;
+            currentKey = null;
+            currentValue = null;
+            return (key, value);
+        }
+
+        char? Peek() => currentPos < input.Length ? input[currentPos] : null;
+
+        bool TryConsume(out char? consumed) {
+            if(input.Length > currentPos) {
+                consumed = input[currentPos];
+                currentPos++;
+                return true;
+            } else {
+                consumed = null;
+                return false;
+            }
+        }
+        
+        void ParseKey() {
+            while (TryConsume(out var c) && c != '=') {
+                currentKey += c;
+            }
+        }
+
+        void ParseQuotedValue() {
+            TryConsume(out var leadingQuote); // consume the leading quote, which we know is there
+            currentValue += leadingQuote;
+            while(TryConsume(out char? c)) {
+                if (c == '"') {
+                    currentValue += c;
+                    // we're done
+                    return;
+                }
+                currentValue += c;
```

Comment:
Use StringBuilder?

Building strings by adding one character at a time is an inefficient quadratic operation. StringBuilder is designed for that.

Reply:
Good spot, I've swapped it out for StringBuilders that get cleared when the key/value pairs are emitted.